### PR TITLE
Use default buildx builder for faster build

### PR DESCRIPTION
**Pull request by KevinMind**
Relates to: mozilla/addons#14972

### Description

Switch to using the default builder for faster builds

### Context

We have been using a custom builder for building our docker image. THis dates back to how we built in circleCI, but the basic point is it is WAY faster to use the default builder on the host.

Essentially it means we don't have to create a new container, share data from the host to that container, and then also export an image back to the host. we can just run the build directly on the host.

Techincally, with the custom builder we have better control over caching and target architecture, but the bake file seems to handle specifying the target architecture and I have verified that caching is working as expected.

### Testing

> NOTE: this is a breaking change. to enable this build even after merged you will have to run

```bash
docker buildx use default
```

The make target removed in this PR `create_docker_builder` effectively told buildx to use the custom builder, even when not specified and so you will continue to use the builder until you manually switch back to the default.

I considered adding logic to automate that, but it's just more code and... well.. if you wantz blazing fast... you typez four words in terminal. 

### Checklist

- [X] Add `#ISSUENUM` at the top of your PR to an existing open issue in the mozilla/addons repository.
- [X] Successfully verified the change locally.
- [X] The change is covered by automated tests, or otherwise indicated why doing so is unnecessary/impossible.
- [ ] Add before and after screenshots (Only for changes that impact the UI).
- [ ] Add or update relevant [docs](../docs/) reflecting the changes made.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -50,6 +50,7 @@ storage/shared_storage/*
 storage/sitemaps/*
 tmp/*
 deps/*
+metadata.json
```
Comment (eviljeff) [inline .gitignore:53]: we're going to need some docs to explain what this metadata.json file is (and a less vague name) - I can see anywhere it's written, only read.

**Review by KevinMind — COMMENTED** [review]
> we're going to need some docs to explain what this metadata.json file is (and a less vague name) - I can see anywhere it's written, only read. 

I've added some doc updates. LMKWYT.

File: .gitignore (hunk 1 above)
Comment (KevinMind) [reply .gitignore:53]: I've added some doc updates. LMKWYT.

Comment (eviljeff) [reply .gitignore:53]: Can we have a more specific name?  metadata.json could be anything.

Comment (KevinMind) [reply .gitignore]: Done.

File: .github/actions/build-docker/action.yml [hunk 2]
```diff
@@ -71,28 +61,25 @@ runs:
         echo "DOCKER_VERSION=${{ steps.meta.outputs.version }}" >> $GITHUB_ENV
         echo "DOCKER_COMMIT=${{ steps.context.outputs.git_sha }}" >> $GITHUB_ENV
         echo "DOCKER_BUILD=${{ steps.context.outputs.git_build_url }}" >> $GITHUB_ENV
+        echo "TAGS_FILE=${{ steps.meta.outputs.bake-file-tags }}" >> $GITHUB_ENV
+        echo "ANNOTATIONS_FILE=${{ steps.meta.outputs.bake-file-annotations }}" >> $GITHUB_ENV
+        echo "DOCKER_METADATA_FILE=buildx-bake-buildx-bake-buildx-bake-metadata.json" >> $GITHUB_ENV
```
Comment (diox) [inline .github/actions/build-docker/action.yml:66]: `buildx-bake-buildx-bake-buildx-bake-metadata.json` ? Why not just `buildx-bake-metadata.json` ?

File: .gitignore [hunk 3]
```diff
@@ -61,7 +62,7 @@ private/
 !deps/.keep
 
 # Ignore buildx bake output files
-buildx-bake-metadata.json
+buildx-bake-buildx-bake-metadata.json
```
Comment (diox) [inline .gitignore:65]: What's this one ?

File: .github/actions/build-docker/action.yml (hunk 2 above)
Comment (KevinMind) [reply .github/actions/build-docker/action.yml:66]: Good catch.

File: .gitignore (hunk 3 above)
Comment (KevinMind) [reply .gitignore:65]: Looks like a broken find/replace

**Review by diox — APPROVED** [review]

**Pull request merged**